### PR TITLE
fix: add build to both CI workflows and adjust dependencies

### PR DESCRIPTION
**Pull request by ryok90**
## Description

Add build job to both CI workflows.  
Add workflow to generate release for this repo.
Improve backporting conditionals for CI workflows.  
Remove check_changes from ci-lambda since it only checks for last commit.

## Is this a breaking change

- [ ] Yes
- [X] No

### CODE REVIEW

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-name-template: '$RESOLVED_VERSION'
-tag-template: '$RESOLVED_VERSION'
+name-template: 'v$RESOLVED_VERSION'
+tag-template: 'v$RESOLVED_VERSION'
```
Comment (MarcioMeier) [inline .github/release-drafter.yml:2]: I thought we had removed the 'v' prefix, we will bring it back?

File: .github/workflows/ci-ui.yml [hunk 2]
```diff
@@ -6,8 +6,17 @@ on:
         required: true
 
 jobs:
+  # Only run CI from unprotected branches (avoid CI for backporting)
+  check_branch:
+    name: Check origin branch
+    if: ${{ (github.head_ref != 'main') && (github.head_ref != 'develop') && !startsWith(github.head_ref, 'release') }}
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo Should run CI based on origin branch? - ${{ steps.should_run.outputs.should_run }}
```
Comment (MarcioMeier) [inline .github/workflows/ci-ui.yml:15]: isn't the step `should_run` missing?

File: .github/release-drafter.yml (hunk 1 above)
Comment (ryok90) [reply .github/release-drafter.yml:2]: We did. This template is only for this repo where the `v` will actually be for better visualization when calling the workflow.
Like this:  
`groundbreaker/workflows/.github/workflows/ci-lambda.yml@v1`

File: .github/workflows/ci-ui.yml (hunk 2 above)
Comment (ryok90) [reply .github/workflows/ci-ui.yml:15]: Yeah, I removed the steps to leave only the conditional on line 12.  
That way if this job skips, all other jobs will skip as well.

Comment (MarcioMeier) [reply .github/workflows/ci-ui.yml:15]: Yeah, but you are using it to print in this step `${{ steps.should_run.outputs.should_run }}`

Comment (ryok90) [reply .github/workflows/ci-ui.yml:15]: oh crap, you're right.
lemme fix it

Comment (ryok90) [reply .github/workflows/ci-ui.yml]: There, moved the conditional to the dependencies job since all others needs this job.  
Thank you Marciooo!

**Review by MarcioMeier — APPROVED** [review]

**Pull request merged**